src/picongpu/CMakeLists.txt

```diff
@@ -219,73 +93,6 @@ find_package(Boost 1.57.0 REQUIRED COMPONENTS program_options regex filesystem
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 set(LIBS ${LIBS} ${Boost_LIBRARIES})
 
-# Boost 1.55 added support for a define that makes result_of look for
```

Comment:
Not the biggest fan of putting this in PMacc only if we use it in PIConGPU `result_of` definitions, too.